src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -467,8 +514,18 @@ primitive_inst::primitive_inst(network& network, program_node const& node, bool
             _outputs = allocate_outputs();
         }
     }
-    if (_impl)
+    if (_impl) {
         _impl->set_node_params(node);
+        if (_impl->is_dynamic()) {
+            _dynamic_impl = _impl->clone();
```

Comment:
Just a comment. 
For multiple streams, we are switching the impl cache to be held by each stream. 
However it seems that dynamic impl does not needed to be held by each stream?

Reply:
But, what if the dyn_impl should be recompiled due to the invalid fusion? 
Then the dyn_impl set to the current prim_inst should be recompiled. 
So ... seems that, dyn_impl should be managed per stream too. ..

Reply:
@yeonbok any implementation needs to be cloned for each stream, as we can't use same kernel object from multiple streams. 
As for invalid fusion, I think current logic doesn't depend on existence of dynamic impl - we just create create dynamic unfused subgraph and compile it. And I assume that this dynamic subgraph may use dynamic impls too

Reply:
- clone: Yes, my comment was about the cache, not the cloining action. 
- unfusion: Yes. the reason I mentioned about unfusion was because I have been thought that using dynamic kernel we dont need to recompile. However if there happens unfusion, then the recompilation happens again...